src/engine/sim_engine/engine.rs

```diff
@@ -40,7 +47,50 @@ impl Engine for SimEngine {
         let redundancy = calculate_redundancy!(redundancy);
 
         match self.pools.get_by_name(name) {
-            Some(_) => Ok(CreateAction::Identity),
+            Some((_, pool)) => {
+                let input_devices = blockdev_paths
+                    .iter()
+                    .cloned()
+                    .collect::<HashSet<&Path, RandomState>>();
+
+                let blockdev_paths: Vec<_> = pool
```

Comment:
Re-use of ```blockdev_paths``` is somewhat confusing. Try something like ```existing_devices``` by analogy to ```input_devices``` instead.

Reply:
Oops! Missed this.

Reply:
Resolved.